### DISCUSSION TITLE
expressions: json to raw

### DISCUSSION THREAD

**Discussion by static-moonlight** [argoproj/argo-workflows#13017] [Q&A]
I have a workflow that is being triggered by a message. This is a simplified version of it:

```json
{
  "data": {
    "files": [
      {
        "name": "path/to/file1",
        "size": 23
      },
      {
        "name": "path/to/file2",
        "size": 42
      }
    ]
  }
}
```

What I need is a flat list of the files names (no json structure):

```
path/to/file1
path/to/file2
```

Currently, I am extracting that list with `jq` in an extra workflow step:

```bash
$ echo ${MESSAGE} | jq --raw-output ".data.files[].name"
```

I was playing around with Argo's expressions but couldn't get the desired result. I tried using `jsonpath` and `map` and also `sprig` function. The closed I got was a json array of the files names. Is there any way to do that with Argo's expressions?

**Comment by wesleyscholl** [comment]
@static-moonlight 

This Argo expression:

```yaml
'{{=join(jsonpath(inputs.parameters.message, "$.data.files[*].name"), "\n")}}'
```

Outputs:
```shell
path/to/file1
path/to/file2
```

See Workflow Variables for more info:

https://argo-workflows.readthedocs.io/en/latest/variables/#expression
